landoui/templates/stack/partials/landing-preview.html

```diff
@@ -43,10 +43,42 @@ <h3 class="StackPage-landingPreview-sectionLabel">Commits:</h3>
         </div>
         <button class="StackPage-landingPreview-expand button"></button>
       </div>
-      <pre class="StackPage-landingPreview-commitMessage">{{
+      {% if revision['should_use_sec_approval_workflow'] %}
+      <div class="StackPage-landingPreview-secureRevisionWarning">
+        <p>
+          <strong>Warning:</strong> This is a secure revision and should follow the
+          <a href="">Security Bug Approval Process</a>.
+        </p>
+        <p>
+          You should consider editing the commit message so that it does not leak
+          security-sensitive information into the Mozilla source repositories.
+        </p>
+        <button class="StackPage-landingPreview-editMessage button">Edit Commit Message</button>
+      </div>
+      {% endif %}
+      <div class="StackPage-landingPreview-displayMessagePanel">
+        <pre class="StackPage-landingPreview-commitMessage">{{
         revision['commit_message']|escape_html|linkify_bug_numbers|linkify_revision_urls|safe
       }}</pre>
-      <div class="StackPage-landingPreview-seeMore"></div>
+        <div class="StackPage-landingPreview-seeMore"></div>
+      </div>
+      <div class="StackPage-landingPreview-editMessagePanel">
+        <form action="/request-sec-approval" method="post">
```

Comment:
I went with the full page redirect after submission because requesting sec-approval modifies the revision and invalidates the stack (the revision gets a blocking reviewer, the stack is now blocked).  I figured redirecting to the confirmation page was an easy way to make sure the stack is refreshed.

Would it be better to show a confirmation modal on the current page, then reload the current page when the user hits "OK"?

I would also add a modal to display form errors.

After submission redirecting to the revision that was changed is definitely a bug.  It should go back to the original stack.

Reply:
@mars-f If you can reload the same page and automatically popup the confirmation modal that would probably work just as well.